lib/api/apiUtils/authorization/permissionChecks.js

```diff
@@ -238,20 +249,21 @@ function checkBucketPolicy(policy, requestType, canonicalID, arn, bucketOwner, l
         const s = copiedStatement[0];
         const principalMatch = _checkPrincipals(canonicalID, arn, s.Principal);
         const actionMatch = _checkBucketPolicyActions(requestType, s.Action, log);
+        const resourceMatch = _checkBucketPolicyResources(request, s.Resource, log);
 
-        if (principalMatch && actionMatch && s.Effect === 'Deny') {
+        if (principalMatch && actionMatch && resourceMatch && s.Effect === 'Deny') {
```

Comment:
The first 3 comparisons are the same for these two if statements so I would merge them.
```js
if ( a && b && c) {
    if ( d ) { blah }
}
```
( cant use the suggestion feature... )

Reply:
Extra question: Why is this using a while loop and not for or something? Looks like a normal for ... array loop but it is instead a while + array thing. Not important just caught my eye.

Reply:
Not sure why it was implemented this way!